src/hooks/useCreateQuiz.js

```diff
@@ -4,45 +4,89 @@ import api from "../config/api";
 import errorDisplayed from "../config/error";
 import useAuth from "./useAuth";
 /* eslint-disable no-unused-vars */
+/* eslint-disable prefer-template */
+/* eslint-disable no-underscore-dangle */
 
 const LOGIN_URL_EXPLORE = "/explore";
 
+// to do: clean urls
+
 export const useCreateQuiz = () => {
   const navigate = useNavigate();
   const { user } = useAuth();
 
   const createQuiz = async (values, onError) => {
     try {
-      const response = await axios.post(
+      const responseCreateQuiz = await axios.post(
         api.quizzes,
         JSON.stringify({ id_user_owner: user.userId,
           title: values.title,
           description: values.description,
-          categories: values.categories,
-          /* questions: values.questions */ }),
+          categories: values.category }),
         {
           headers: { Authorization: `Bearer ${user.token}`,
             "Content-Type": "application/json" },
           withCredentials: true
         }
       );
 
-      /* console.log(response.data);
-      console.log(JSON.stringify(response)); */
+      console.log("responseCreateQuizSkeleton", responseCreateQuiz);
+
+      const convertedQuestions = [];
+      values.questions.map((question, position) => convertedQuestions.push({
+        id_quiz: responseCreateQuiz.data.idQuiz,
+        title: question.question,
+        choices: [],
+        explanation: question.explanation,
+        link_to_learn_more: question.learnMore,
+        position
+      }));
+
+      console.log("convertedQuestions", convertedQuestions);
+      const responseCreateQuestions = await axios.post(
+        "quizzes/" + responseCreateQuiz.data.idQuiz + "/questions",
```

Comment:
By 'generation des urls (cleancode)' did you mean stock this path in the variable?

Reply:
I meant create of function that will generate path dynamically with the right idQuiz and IdQuestion when need. I have created one this morning it's in the api file => `export const generateRoute = (route, quizId, questionId) => generatePath(route, { quizId, questionId });` maybe the name generateRoute is not accurate but I am using generatePath...

Reply:
j'ai fait des tests, tout semble marcher correctement, bravo ! J'aimerais bien que tu expliques un peu plus ça pendant le daily après ;)